libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -434,6 +434,23 @@ bool AP_Mission::get_next_nav_cmd(uint16_t start_index, Mission_Command& cmd)
     return false;
 }
 
+/// get_next_nav_cmd_id - gets next "navigation" command found at or after current command
```

Comment:
We do have a `is_takeoff_next` and  `starts_with_takeoff_cmd`. I think you would still need a little special handling due to the "treat FW takeoff as VTOL" option but that may be a better option,

Reply:
it needed some tweaks for VTOL_TAKEOFF, needs changes in copter as well